src/runtime/pymodule.cs

```diff
@@ -37,5 +47,39 @@ public static PyModule FromString(string name, string code)
             PythonException.ThrowIfIsNull(m);
             return new PyModule(ref m);
         }
+
+        public static bool IsInSysModules(string name)
+        {
+            // first check if there is an existing module
+            BorrowedReference modules = Runtime.PyImport_GetModuleDict();
+            BorrowedReference m = Runtime.PyDict_GetItemString(modules, name);
+            return !m.IsNull && Runtime.PyObject_TypeCheck(m, new BorrowedReference(Runtime.PyModuleType));
+        }
+
+        public static PyModule Create(string name, string filename=null)
+        {
+            NewReference op = Runtime.PyModule_New(name);
+            PythonException.ThrowIfIsNull(op);
+
+            if (filename != null)
+            {
+                BorrowedReference globals = Runtime.PyModule_GetDict(new BorrowedReference(op.DangerousGetAddress()));
+                PythonException.ThrowIfIsNull(globals);
+                int rc = Runtime.PyDict_SetItemString(globals, "__file__", new BorrowedReference(filename.ToPython().Handle));
+                PythonException.ThrowIfIsNotZero(rc);
+            }
+
+            return new PyModule(ref op);
+        }
```

Comment:
I suggest to make `Create` method private, and add a constructor instead. E.g.
`public PyModule(string name, string filename = null): this(Create(name, filename)) { }`